lochness/transfer/__init__.py

```diff
@@ -0,0 +1,296 @@
+                      timestamp
+                      1621470407.030176
+                      1621470455.6350138
+    Returns:
+        last_compress_timestamp: the timestamp for the last data compression.
+                                 If the compress_db does not exist, a pseudo-
+                                 random timestamp is returned to execute sync
+                                 for all of the data under PHOENIX.
+        now: timestamp for the current time
+        compress_df: table with the history of data compression including
+                     the current time stamp.
+    '''
+    tmp_timestamp = 590403600
+    if Path(timestamp_db).is_file():
+        compress_df = pd.read_csv(timestamp_db, index_col=0)
+        try:
+            last_compress_timestamp = compress_df['timestamp'].max()
+        except KeyError:
+            last_compress_timestamp = tmp_timestamp
+    else:
+        compress_df = pd.DataFrame({'timestamp': [tmp_timestamp]})
+        last_compress_timestamp = tmp_timestamp
+
+    # get current time and update the database
+    now = time()
+    compress_df = pd.concat([compress_df,
+                             pd.DataFrame({'timestamp': [now]})])
+
+    return last_compress_timestamp, now, compress_df
+
+
+def compress_new_files(compress_db: str, phoenix_root: str,
+                       out_tar_ball: str, general_only: bool = True) -> None:
+    '''Find a list of new files from the last lochness to lochness sync
+
+    Key arguments:
+        compress_db: path of a csv file, which contains timestamp of each
+                     lochness to lochness sync attempt (compression), str.
+                     eg) lochness_sync_history.csv
+                     timestamp
+                     1621470407.030176
+                     1621470455.6350138
+        phoenix_root: PHOENIX root, str.
+        out_tar_ball: compressed tarball output path, str.
+        general_only: only compress the data under GENERAL if True, bool.
+
+    Returns:
+        None
+    '''
+    last_compress_timestamp, now, compress_df = get_ts_and_db(compress_db)
+
+    # find new files and zip them
+    new_file_lists = get_updated_files(phoenix_root,
+                                       last_compress_timestamp,
+                                       now,
+                                       general_only)
+
+    compress_list_of_files(phoenix_root, new_file_lists, out_tar_ball)
+
+    # save database when the process completes
+    compress_df.to_csv(compress_db, index=False)
+
+
+def send_data_over_sftp(Lochness, file_to_send: str):
+    '''Send data over sftp'''
+
+    sftp_keyring = Lochness['keyring']['lochness_to_lochness']
+    host = sftp_keyring['HOST']
+    username = sftp_keyring['USERNAME']
+    password = sftp_keyring['PASSWORD']
+    path_in_host = sftp_keyring['PATH_IN_HOST']
+    port = sftp_keyring['PORT']
+
+    transport = paramiko.Transport((host, int(port)))
```

Comment:
In my experience, anything in Python that is wrapped around a shell command is hard to debug. I also suggested shell command for this purpose in Teams. It could be something like:

> subprocess.Popen("scp tar.ball user@host -p port", shell=True)

Reply:
Paramiko module is already used in the original `lochness` (harvard-nrg). It is a module that is well maintained, and also easy to use. I think there are even exceptions already made available in paramiko, which we may add in a future version.